cluster-autoscaler/cloudprovider/aws/aws_cloud_provider_test.go

```diff
@@ -603,7 +602,7 @@ func TestDeleteNodesWithPlaceholder(t *testing.T) {
 	err = asgs[0].DeleteNodes([]*apiv1.Node{node})
 	assert.NoError(t, err)
 	a.AssertNumberOfCalls(t, "SetDesiredCapacity", 1)
-	a.AssertNumberOfCalls(t, "DescribeAutoScalingGroupsPages", 1)
+	a.AssertNumberOfCalls(t, "DescribeAutoScalingGroupsPages", 2)
```

Comment:
I think we need another test here that verifies the behaviour we're expecting -- specifically, create an ASG with (say) 7 instances and a desired capacity of (say) 10, and then call `DeleteNodes` with 5 placeholders and 2 real nodes.  Then we should see that the target size of the ASG goes down to 5.

Reply:
Done. I have added context on the unit test to avoid any confusion.